fabric-fluid-v1/src/main/java/net/fabricmc/fabric/api/fluids/v1/FluidView.java

```diff
@@ -0,0 +1,132 @@
+ *
+ * @see Drops
+ */
+public class FluidView {
+	private FluidView() {
+	}
+
+	/**
+	 * get the fluid container for the given stack.
+	 *
+	 * @param stack the itemstack
+	 * @param sink a place for the item to put it's byproducts into, and take items when being filled
+	 * @return the fluid container for the stack
+	 */
+	public static FluidContainer getFluidContainer(ItemStack stack, ItemSink sink) {
+		Item item = stack.getItem();
+
+		if (item instanceof ItemFluidContainer) {
+			return ((ItemFluidContainer) item).getContainer(sink, stack);
+		}
+
+		return ImmutableFluidVolume.EMPTY;
+	}
+
+	/**
+	 * merge all the entity and block containers in the given location for the given face into one fluid container.
+	 *
+	 * @param world the world
+	 * @param pos the position
+	 * @param face the face
+	 * @return a fluid container for the face
+	 */
+	public static FluidContainer getFluidContainer(World world, BlockPos pos, Direction face) {
+		FluidContainer container = getBlockFluidContainer(world, pos, face);
+		Iterable<FluidContainer> entities = getEntityFluidContainersIterable(world, pos, face);
+
+		if (Iterables.isEmpty(entities)) {
+			return container;
+		}
+
+		if ((container.isImmutable() && container.isEmpty())) {
+			return new SimpleFluidContainer(entities);
+		} else {
+			return new SimpleFluidContainer(Iterables.concat(() -> Iterators.singletonIterator(container), getEntityFluidContainersIterable(world, pos, face)));
+		}
+	}
+
+	/**
+	 * get the fluid container for the block at the given location at the given side.
+	 *
+	 * @param world the world
+	 * @param pos the position
+	 * @param face the face to access
+	 * @return the fluid container
+	 */
+	public static FluidContainer getBlockFluidContainer(World world, BlockPos pos, Direction face) {
+		BlockState state = world.getBlockState(pos);
+		Block block = state.getBlock();
+
+		if (block instanceof SidedFluidContainer) {
+			return ((SidedFluidContainer) block).getContainer(world, pos, face);
+		}
+
+		return ImmutableFluidVolume.EMPTY;
+	}
+
+	private static Iterable<FluidContainer> getEntityFluidContainersIterable(World world, BlockPos pos, Direction face) {
+		List<Entity> entities = world.getEntities((Entity) null, new Box(pos), e -> e instanceof SidedFluidContainer);
+
+		if (entities.isEmpty()) {
+			return Collections::emptyIterator;
+		}
+
+		return () -> new Iterator<FluidContainer>() {
+			private Iterator<Entity> iterator = entities.iterator();
+
+			@Override
+			public boolean hasNext() {
+				return this.iterator.hasNext();
+			}
+
+			@Override
+			public FluidContainer next() {
+				return ((SidedFluidContainer) this.iterator.next()).getContainer(world, pos, face);
+			}
+		};
+	}
+
+	public static FluidContainer getEntityFluidContainer(World world, BlockPos pos, Direction face) {
+		return new SimpleFluidContainer(getEntityFluidContainersIterable(world, pos, face));
+	}
+
+	public static boolean miscible(Fluid a, Fluid b) {
```

Comment:
I now understand this is an actyal word, but can you document what on earth `miscible` means or the context it is used within.

Reply:
I'm going to rename it to something normal